### ISSUE TITLE
Feature: Training data fuzzing (noise injection) to prevent memorisation (#1899)

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1900]
## Summary

Implement automatic noise injection (fuzzing) on training data during the training loop to prevent networks from memorising exact training examples. This is a standard regularisation technique that forces the network to learn general patterns rather than exact data points.

### 1. Create `DataFuzzingConfig.ts`

Create `src/config/DataFuzzingConfig.ts` following the established config pattern (`CrossValidationConfig.ts` is a good reference):

- `DataFuzzingConfig` — optional fields interface
- `RequiredDataFuzzingConfig` — `Required<DataFuzzingConfig>`
- `DEFAULT_DATA_FUZZING_CONFIG` — default values

**Configuration fields:**

| Field | Type | Default | Range | Description |
|-------|------|---------|-------|-------------|
| `enabled` | `boolean` | `false` | — | Whether data fuzzing is active |
| `inputNoiseScale` | `number` | `0.01` | `0..1` | Standard deviation (Gaussian) or half-width (uniform) of noise added to inputs |
| `outputNoiseScale` | `number` | `0` | `0..1` | Optional noise on target outputs (0 = disabled). Small values provide label smoothing. |
| `noiseType` | `"gaussian" \| "uniform"` | `"gaussian"` | — | Distribution used for noise generation |

### 2. Wire Config into the Configuration System

Following the [config pattern](https://github.com/stSoftwareAU/NEAT-AI/blob/Develop/src/config/CrossValidationConfig.ts):

1. Add `dataFuzzing: RequiredDataFuzzingConfig` to `NeatArguments.ts`
2. Add `dataFuzzing?: DataFuzzingConfig` to `NeatOptions.ts` (both `NeatOptions` and `NeatOptionsInput` types, with `CoerceNumeric<>` for CLI numeric fields)
3. Add `dataFuzzing` to both `Omit` lists in `NeatOptions.ts`
4. Parse in `NeatConfig.ts` / `NeatConfigParsers.ts` using `parseNumber()` with IIFE pattern
5. Pass through to `TrainOptions.ts` / `TrainArguments` so the training loop can access it

### 3. Implement Noise Injection in Training Loop

In `src/architecture/Training.ts`, function `trainDirBinary()`:

**Integration point** — after copying record data into pre-allocated buffers (lines ~399-400) and **before** `activateAndTrace()`:

```typescript
// After:
observationsBuffer.set(recordArray.subarray(0, creature.input));
targetsBuffer.set(recordArray.subarray(creature.input));

// Apply fuzzing if enabled:
if (fuzzingConfig.enabled) {
  applyNoise(observationsBuffer, fuzzingConfig.inputNoiseScale, fuzzingConfig.noiseType, rng);
  if (fuzzingConfig.outputNoiseScale > 0) {
    applyNoise(targetsBuffer, fuzzingConfig.outputNoiseScale, fuzzingConfig.noiseType, rng);
  }
}
```

Create a utility function (e.g., `src/propagate/DataFuzzing.ts` or similar) that applies noise in-place to a `Float32Array`:
- **Gaussian**: Add `scale * rng.gaussian()` to each element (Box-Muller or similar)
- **Uniform**: Add `scale * (2 * rng.random() - 1)` to each element

**Important**: Noise must be freshly generated per record per iteration (not cached) to prevent the network from learning the noise pattern.

### 4. Extend to Predictive Coding Trainer

The `trainWithPredictiveCoding()` path in `PredictiveCodingTrainer.ts` has its own data reading loop. Apply the same fuzzing there so that all training modes benefit.

### 5. Tests

Add tests in `test/` directory (mirroring `src/` structure):
- Unit tests for the noise application function (verify statistical properties — mean near zero, std dev matches scale)
- Unit tests for config parsing (verify defaults, bounds, validation)
- Integration test: train a small network with fuzzing enabled and verify it completes without error
- Verify that fuzzing is NOT applied during inference (only during training)

## Acceptance Criteria

- [ ] `DataFuzzingConfig.ts` follows the established three-type config pattern
- [ ] Config is wired into `NeatArguments`, `NeatOptions`, and `NeatConfig`
- [ ] Noise is injected per-sample in the training loop, freshly generated each time
- [ ] Both Gaussian and uniform noise types work correctly
- [ ] Input noise and output noise can be controlled independently
- [ ] Disabled by default (backward compatible — no behaviour change unless opted in)
- [ ] Works with both standard backpropagation and predictive coding training paths
- [ ] Cross-validation path automatically inherits fuzzing (via `trainDirBinary`)
- [ ] All tests pass, `./quality.sh` succeeds
- [ ] Australian English spelling throughout (e.g., "behaviour", "normalise")

## Context

- This is a standard regularisation technique (also called "input perturbation" or "noise injection")
- It works by making it impossible for the network to memorise exact input-output mappings
- Complements existing regularisation: dropout (#1860), L1/L2 decay (#1859), cross-validation (#1865)
- The RNG should use the existing `getRandomNumberGenerator()` for reproducibility when seeded

Part of #1899

**Comment by stservice** [comment]
Claimed by `GRQ-23`

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/1902 has been created to address this issue.

**Comment by stservice** [comment]
Automatically closed — PR #1902 has been merged.
